src/proj.h

```diff
@@ -0,0 +1,205 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *****************************************************************************/
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#endif
+#include <math.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#ifndef PROJ_H
+#define PROJ_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/******************************************************************************
+    proj.h is included by projects.h in order to determine the size of the
+    OBSERVATION object.
+
+    In order to stomp as little as possible on the traditional proj.4 name
+    space, proj.h is littered with inclusion guards, leaving only the minimum
+    possible implementation when included from projects.h
+
+    The OBSERVATION object is fully defined if proj.h is included alone or
+    in connection with *but before* projects.h (the latter may be needed in
+    some cases, where it is necessary to access this "bare essentials" API,
+    while still having direct access to PJ object internals)
+******************************************************************************/
+
+/* Data type for generic geodetic observations */
+struct OBSERVATION;
+typedef struct OBSERVATION OBSERVATION;
+
+/* Data type for generic geodetic 3D data */
+union PJ_TRIPLET;
+typedef union PJ_TRIPLET PJ_TRIPLET;
+
+/* Data type for generic geodetic 3D data plus epoch information */
+union PJ_SPATIOTEMPORAL;
+typedef union PJ_SPATIOTEMPORAL PJ_SPATIOTEMPORAL;
+
+#ifndef PROJECTS_H
+/* Data type for projection/transformation information */
+struct PJconsts;
+typedef struct PJconsts PJ;         /* the PJ object herself */
+
+/* Omega, Phi, Kappa: Rotations */
+typedef struct {double o, p, k;}  OPK;
+
+/* Easting, Northing, and some kind of height (orthometric or ellipsoidal) */
+typedef struct {double e, n, h;}  ENH;
+
+/* Geodetic spatiotemporal coordinate types */
+typedef struct { double   x,   y,  z, t; }  XYZT;
+typedef struct { double   e,   n,  h, t; }  ENHT;
+typedef struct { double   u,   v,  w, t; }  UVWT;
+typedef struct { double lam, phi,  z, t; }  LPZT;
+
+/* Classic proj.4 pair/triplet types */
+typedef struct { double   u,   v; }  UV;
+typedef struct { double   x,   y; }  XY;
+typedef struct { double lam, phi; }  LP;
+
+typedef struct { double   x,   y,  z; }  XYZ;
+typedef struct { double   u,   v,  w; }  UVW;
+typedef struct { double lam, phi,  z; }  LPZ;
+
+/* Ancillary pairs and triplets for geodetic computations */
+
+/* Degrees, minutes, and seconds */
+typedef struct { double d, m,  s; }  DMS;
+
+/* Geoid undulation and deflections of the vertical */
+typedef struct { double   N,   z,  e; }  PJ_ANC_GEOIDAL;
```

Comment:
Ancillary data. There's a comment about it a few lines up - I have now also added some inline comments, and changed the PJ_ANC_* names for consistency